diff --git a/BUILD.gn b/BUILD.gn
index 7168ea2da541ef11090155c04ed05e5ddf351f55..94034ffbce1aba43c4ac3af6cf3414ffe4b375b8 100644
--- a/BUILD.gn
+++ b/BUILD.gn
@@ -179,7 +179,13 @@ config("internal_config") {
 
   include_dirs = [ "." ]
 
-  if (is_component_build) {
+  if (is_win) {
+    cflags = [
+      "/wd4275",  # dll export
+    ]
+  }
+
+  if (true) {
     defines = [ "BUILDING_V8_SHARED" ]
   }
 }
diff --git a/src/profiler/profiler-listener.cc b/src/profiler/profiler-listener.cc
index 4609f4a3f4df039037883e63365f47225c95851c..56a191189e4a3c4b0b8ac2fff9c0735b02b23213 100644
--- a/src/profiler/profiler-listener.cc
+++ b/src/profiler/profiler-listener.cc
@@ -97,7 +97,7 @@ void ProfilerListener::CodeCreateEvent(CodeEventListener::LogEventsAndTags tag,
       int position = it.source_position().ScriptOffset();
       int line_number = script->GetLineNumber(position) + 1;
       int pc_offset = it.code_offset() + offset;
-      line_table->SetPosition(pc_offset, line_number);
+      line_table->SetPosition(pc_offset, line_number > 0 ? line_number : line);
     }
   }
   rec->entry = NewCodeEntry(
